includes/wc-template-functions.php

```diff
@@ -1976,7 +1977,10 @@ function woocommerce_get_product_subcategories( $parent_id = 0 ) {
 				'taxonomy'     => 'product_cat',
 				'pad_counts'   => 1,
 			) ) );
-			wp_cache_set( 'product-categories-' . $parent_id, $product_categories, 'product_cat' );
+
+			$cache[ $parent_id ] = $product_categories;
+
+			wp_cache_set( 'product-category-hierarchy', $cache, 'product_cat' );
```

Comment:
Since we are storing data for all product categories in a single cache entry, I wonder if its size might be a problem for stores with a significant number of product categories (or long descriptions for each category or something else that will impact the size of the `$cache` variable). Thoughts?

On WordPress.com servers Memcached objects must be smaller than 1MB. If they are bigger than the limit, they are not cached, and the code will hit the database to get the data.